modules/geo-layers/src/tile-layer/tile-layer.js

```diff
@@ -57,28 +57,42 @@ export default class TileLayer extends CompositeLayer {
       changeFlags.dataChanged ||
       (changeFlags.updateTriggersChanged &&
         (changeFlags.updateTriggersChanged.all || changeFlags.updateTriggersChanged.getTileData));
-
     if (createTileCache) {
-      if (tileset) {
-        tileset.finalize();
-      }
-      tileset = new Tileset2D({
+      const opts = {
         ...this._getTilesetOptions(props),
         getTileData: this.getTileData.bind(this),
         onTileLoad: this._onTileLoad.bind(this),
         onTileError: this._onTileError.bind(this),
         onTileUnload: this._onTileUnload.bind(this)
+      };
+      let reload = false;
+      // If there is a tileset already, update it and force a reload.
+      if (tileset) {
+        reload = true;
+        tileset.finalize();
+        tileset.loadOptions(opts);
+        this._updateTileset(tileset, reload);
+        // Otherwise, we need to create a new tileset.
+      } else {
+        tileset = new Tileset2D(opts);
+        this._updateTileset(tileset, reload);
+        this.setState({tileset});
+      }
+      // Force an update of the tile layers without deleting them to prevent skipped frames.
+      tileset.tiles.forEach(tile => {
+        tile.updateLayers = true;
       });
-      this.setState({tileset});
     } else if (changeFlags.propsChanged || changeFlags.updateTriggersChanged) {
       tileset.setOptions(this._getTilesetOptions(props));
       // if any props changed, delete the cached layers
-      this.state.tileset.tiles.forEach(tile => {
+      // Should we do the same as the above?  Not delete but use a boolean to cause an update?
+      tileset.tiles.forEach(tile => {
```

Comment:
@Pessimistress Not sure what the origin of this behavior was but seems similar to the above?  We should just set them to have `updateLayers` be `true`